### PR TITLE
Task3

### PR DESCRIPTION
**Pull request by HalinaPP**
- [x] Frontend application is integrated with product service (/products API) and products from product-service are represented on Frontend.
- [x] The getProductsList (https://tgyctvcbw8.execute-api.eu-west-1.amazonaws.com/dev/products/) lambda function returns a correct response 
- [x] The  getProductsById(https://tgyctvcbw8.execute-api.eu-west-1.amazonaws.com/dev/products/7567ec4b-b10c-48c5-9345-fc73c48a80aa or another from getProductsList )  lambda function returns a correct response 

**Additional scope**
- [x] Async/await is used in lambda functions
- [x] ES6 modules are used for product-service implementation
- [x] Webpack is configured for product-service
- [x] SWAGGER documentation is created for product-service (https://app.swaggerhub.com/apis-docs/HalinaPP/product-service/1.0.0#/Product)
- [x] Lambda handlers (getProductsList, getProductsById) code is written not in 1 single module (file) and separated in codebase.
- [x] Main error scenarious are handled by API ("Product not found" error, try catch blocks are used in lambda handlers).

**Link to product-service API** - https://tgyctvcbw8.execute-api.eu-west-1.amazonaws.com/dev/products/

LInk to FE MR - https://d31o38wsvksq0v.cloudfront.net/
https://github.com/HalinaPP/nodejs-aws-fe/pull/2

### CODE REVIEW

File: product-service/handlers/getProductsById.js [hunk 1]
```diff
@@ -1,13 +1,31 @@
 'use strict';
+const data = require('../data/mock.json');
 
 module.exports.getProductsById = async event => {
+  //try{
+    const productId = event.path.split('/')[2];
```
Comment (izzetbek) [inline product-service/handlers/getProductsById.js:6]: Тут можно достать из event.pathParameters.id

File: product-service/handlers/getProductsList.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    if(!data){
+      throw new Error( 'Error: products not found1!' );
+    }
+
+    return {
+        statusCode: 200,
+        headers: {
+          'Access-Control-Allow-Origin': '*',
+          'Access-Control-Allow-Credentials': true,
+        },
```
Comment (spiderpoul) [inline product-service/handlers/getProductsList.js:16]: you could create a util to process response and doesn't write headers and status every time, like:
```
 data =>   ({
       statusCode: 200,
        headers: {
          'Access-Control-Allow-Origin': '*',
          'Access-Control-Allow-Credentials': true,
        },
        body: JSON.stringify(data)
    })
```

File: product-service/handlers/getProductsById.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
+'use strict';
+import data from '../data/mock.json';
+
+export const getProductsById = async event => {
+  try{ 
+    const productId = event.path.split('/')[2];
+  
+    const product = await data.find(item => item.id === productId);
+  
+      if(!product){
```
Comment (aliakseikrauchanka) [inline product-service/handlers/getProductsById.js:10]: Норм, можно чуть лучше - сразу возвращать респонс с 404 кодом без try catch блока

File: product-service/handlers/getProductsList.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
+'use strict';
+import data from '../data/mock.json';
+
+export const getProductsList = async event => {
+
+  try{
+    if(!data){
```
Comment (aliakseikrauchanka) [inline product-service/handlers/getProductsList.js:7]: В нашем конкретном случае мы точно знаем что продукты есть, поэтому похоже необязательно проверять на их отсутсвие и кидать ошибку тут :)

File: product-service/handlers/getProductsList.js [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+        headers: {
+          'Access-Control-Allow-Origin': '*',
+          'Access-Control-Allow-Credentials': true,
+        },
+        body: JSON.stringify(data, null, 2)
+      
+      };
+    } catch (e) {
+      return {
+        statusCode: 400,
```
Comment (aliakseikrauchanka) [inline product-service/handlers/getProductsList.js:22]: Если кидать ошибку на отсутствие, то по HTTP лучше будет подходить 404 нежеле 400

**Pull request merged**